### DISCUSSION TITLE
Circular endpoint call timeout between FastApi services

### DISCUSSION THREAD

**Discussion by georgios-v** [fastapi/fastapi#8638] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import requests

app = FastAPI()


@app.get("/A1")
def read_root():
    return requests.get(url="http://localhost/B1")


@app.get("/A2")
def read_root():
    pass

@app.get("/B1")
def read_root():
     return requests.get(url="http://localhost/A2")  # this call will timeout
```

### Description

Everything is tested only using development mode with auto reload enabled.

Assume services A and B both implemented using FastApi
Service A has endpoints A.1 and A.2
Service B has endpoint B.1

an external request arrives to endpoint A.1 which makes a call to endpoint B.1 (requests.post) which makes a call to endpoint A.2 (requests.delete). The final call times out resulting in the initial call to end in 500!
the handlers of each endpoint are simple CRUD operations

I know that design changes can remove the circular calls and everything will function. However, I'd really like to get to the bottom of this issue and understand why it is happening

thanks

### Operating System

Linux

### FastAPI Version

0.54.2

### Python Version

3.7.7

**Comment by Trinkes** [comment]
Hello @georgios-v 

how are you running your services?
I had the same issue in the past and that was caused because I only had 1 worker.

If that's your case, here's what is happening:
1. an external request arrives to endpoint A.1
2. A.1 endpoint makes the request to B.1 
3. fast api(or starlette i'm not sure) puts the B.1 request on hold while waiting for a worker to dispatch it
4. Time out happens because the existing only worker is processing A.1 request.

Solution:
Run the api with multiple workers
if you're using uvicorn, use the [--workers](https://www.uvicorn.org/deployment/) argument

Example: `$ uvicorn example:app --workers 4`

**Comment by YuriiMotov** [comment]
Can't reproduce this. Works fine in current version after fixing obvious error:

```python
return requests.get(url="http://localhost:8000/A2").text  # `.text` was added
```
